src/ext/snp/mod.rs

```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
     pub report: &'a [u8],
 }
 
+flags! {
+    pub enum Flags: u8 {
+        /// Indicates if only one socket is permitted
+        SingleSocket = 0b00010000u8,
```

Comment:
```suggestion
        SingleSocket = 1 << 4,
```

This is more idiomatic and easier to read. It says that `SingleSocket is the fifth bit` (i.e. 4, but zero-indexed).